include/s3select_functions.h

```diff
@@ -643,6 +647,40 @@ struct _fn_substr : public base_function{
     
 };
 
+struct _fn_lower : public base_function {
+
+    value v_str;
+    std::string buff;
+
+    bool operator()(std::vector<base_statement*> * args,variable * result)
+    {
+        std::vector<base_statement*>::iterator iter = args->begin();
+        base_statement* str =  *iter;
+        v_str = str->eval();
```

Comment:
its possible to avoid the assignment operator (value operator = )
it construct v_str per each call , no need for that.